### DISCUSSION TITLE
Cad Viewer has a problem on a specific layer

### DISCUSSION THREAD

**Discussion by melihpeker-pointr** [mozman/ezdxf#475] [Issues]
Hi everyone, 

I had a DXF file to view and I observed an unusual behavior. When I tried to view that file as it is, the resulting image is like below (very very far away):

![far away](https://user-images.githubusercontent.com/83574547/122575984-eb1ac900-d059-11eb-9215-2404e35c8daa.png)

Then I realised that a specific layer causes that problem. When I tried to view that layer using AutoCAD, result is below:
<img width="307" alt="Screenshot 2021-06-18 at 17 24 32" src="https://user-images.githubusercontent.com/83574547/122576146-0dace200-d05a-11eb-9ee0-06cef25a2e25.png">

However, when I use Cad Viewer to view that layer, I can't see anything. Moreover, when I disable that layer on the original DXF, I can view the rest of the file without any problem using Cad Viewer. Can you check what is wrong with that layer?

The problematic layer: [Problem.zip](https://github.com/mozman/ezdxf/files/6677597/Problem.zip)

Thanks in advance!

**Comment by mbway** [comment]
this is a very common issue for some reason. Many CAD files that I deal with are drawn carelessly with lots of junk surrounding the actual content, often very far away like you have encountered.

I don't know how CAD applications determine what the view limits should be. I assume they must be in the dxf somewhere because opening the file in autocad shows:
![image (47)](https://user-images.githubusercontent.com/4923501/122580711-2911f000-d04e-11eb-95b4-4ebbf194ba9b.png)


The solution for ezdxf would be to set the view limits manually like in https://github.com/mozman/ezdxf/discussions/451 or remove the junk entities by some criteria. For my own uses I have an algorithm which does this by looking for 'outlier' entities to remove.

For this particular case, if the view limits are correct when a single layer is excluded, you could draw the document without that layer, record the view limits then draw the whole document then restore the limits

**Reply by melihpeker-pointr** [reply]
How do you remove the 'outlier' entities? Do you first find the ROI in the drawing and then remove the rest?

**Reply by melihpeker-pointr** [reply]
Btw, I may found a solution for that. When I explode the blocks into poly lines and also explode the poly lines, I can manage to view the Problem.dxf as it is shown in AutoCAD.
